app/main.py

```diff
@@ -31,17 +29,27 @@ def __mul__(self, other: int | float) -> Distance:
     def __truediv__(self, other: int | float) -> Distance:
         return Distance(distance=round(self.km / other, 2))
 
-    def __eq__(self, other: int | float) -> bool:
-        return self.km == other
+    def __eq__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other.real
 
-    def __lt__(self, other: int | float) -> bool:
-        return self.km < other
+    def __lt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km < other.km
+        return self.km < other.real
 
-    def __gt__(self, other: int | float) -> bool:
-        return self.km > other
+    def __gt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km > other.km
+        return self.km > other.real
 
-    def __le__(self, other: int | float) -> bool:
-        return self.km <= other
+    def __le__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        return self.km <= other.real
 
-    def __ge__(self, other: int | float) -> bool:
-        return self.km >= other
+    def __ge__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, Distance):
+            return self.km >= other.km
+        return self.km >= other.real
```

Comment:
You didn't fix previous comment:

Refactor all functions using only 1 `return`. Operator `not` might help you in this case, use it.

Reply:
Example:

```
return not self.km < other
```